examples/07_advanced/plot_mask_large_fmri.py

```diff
@@ -0,0 +1,414 @@
+
+from nilearn.datasets import fetch_adhd
+from nilearn.image import concat_imgs
+
+N_SUBJECTS = 6
+N_REGIONS = 6
+
+fmri_data = fetch_adhd(n_subjects=N_SUBJECTS)
+fmri_img = concat_imgs(fmri_data.func)
+
+output_dir = Path.cwd() / "results" / "plot_mask_large_fmri"
+output_dir.mkdir(parents=True, exist_ok=True)
+print(f"Large fmri file will be saved to:\n{output_dir}")
+
+fmri_path = output_dir / "large_fmri.nii.gz"
+fmri_img.to_filename(fmri_path)
+
+# %%
+# Create a set of binary masks
+# ----------------------------
+# We will now create 6 binary masks from a brain atlas. Here we will use the
+# multiscale functional brain parcellations via the
+# :func:`~nilearn.datasets.fetch_atlas_basc_multiscale_2015` function.
+# We will fetch a 64-region version of this atlas and then create separate
+# binary masks for the first 6 regions.
+
+from nilearn.datasets import fetch_atlas_basc_multiscale_2015
+from nilearn.image import load_img, new_img_like, resample_to_img
+
+atlas_path = fetch_atlas_basc_multiscale_2015(resolution=64).maps
+
+atlas_img = load_img(atlas_path)
+
+# %%
+# In this case, the atlas and the fMRI image have different resolutions.
+# So we will resample the atlas to the fMRI image. It is important to do that
+# because only :class:`~nilearn.maskers.NiftiMasker` can handle the resampling
+# of the mask to the fMRI image but other methods considered here will not.
+
+resampled_atlas = resample_to_img(
+    atlas_img,
+    fmri_img,
+    interpolation="nearest",
+    copy_header=True,
+    force_resample=True,
+)
+
+mask_imgs = []
+mask_paths = []
+for idx in range(N_REGIONS):
+    mask = resampled_atlas.get_fdata() == idx
+    mask = new_img_like(
+        ref_niimg=fmri_img,
+        data=mask,
+        affine=resampled_atlas.affine,
+        copy_header=True,
+    )
+
+    mask_imgs.append(mask)
+    path = output_dir / f"mask_{idx}.nii.gz"
+    mask.to_filename(path)
+    mask_paths.append(path)
+
+# %%
+# Mask the fMRI image using NiftiMasker
+# -------------------------------------
+# Let's first see how we would typically mask the fMRI image using
+# :class:`~nilearn.maskers.NiftiMasker`. This is the most common way to extract
+# data from an fMRI image as it makes it easy to standardize, smooth, detrend,
+# etc. the data.
+#
+# We will define a function that will do that so that it's easier to use it
+# with the ``memory_profiler`` package to measure the memory usage.
+
+from nilearn.maskers import NiftiMasker
+
+
+def nifti_masker_single(fmri_path, mask_path):
+    return NiftiMasker(mask_img=mask_path).fit_transform(fmri_path)
+
+
+# %%
+# Furthermore, we can input the fmri image and the masks in two different ways:
+#
+# 1. Using the file paths
+# 2. Using the in-memory objects
+#
+# So we will measure the memory usage for both cases.
+#
+# Let's first create a dictionary to store the memory usage for each method
+
+from memory_profiler import memory_usage
+
+nifti_masker = {"single": {"path": [], "in_memory": []}}
```

Comment:
Maybe let's define the dictionary all at once rather than iteratively ? I don't think we've explained why we expect `parallel` to be different than `single` in any of the text to date ; this might be a good place to state that directly. 

```suggestion
nifti_masker = {
    "single": {"path": [], "in_memory": []},
    "parallel": {"path": [], "in_memory": []}
}
```

Reply:
I'm not storing the memory usage for single cases anymore. So this isn't needed